cmd/protoc-gen-gorums/gengorums/gorums.go

```diff
@@ -129,20 +176,40 @@ var gorumsTypes = map[string]*protoimpl.ExtensionInfo{
 	gorums.E_Multicast.Name[index:]:         gorums.E_Multicast,
 }
 
+// name to strict ordering type mapping
+var strictOrderingTypes = map[string]*protoimpl.ExtensionInfo{
+	strictordering.E_StrictOrderingQc.Name[soIndex:]:  strictordering.E_StrictOrderingQc,
+	strictordering.E_StrictOrderingRpc.Name[soIndex:]: strictordering.E_StrictOrderingRpc,
+}
+
 var gorumsCallTypeTemplates = map[*protoimpl.ExtensionInfo]string{
-	gorums.E_Qc:                quorumCall,
-	gorums.E_QcFuture:          futureCall,
-	gorums.E_Correctable:       correctableCall,
-	gorums.E_CorrectableStream: correctableStreamCall,
-	gorums.E_Multicast:         multicastCall,
+	gorums.E_Qc:                        quorumCall,
+	gorums.E_QcFuture:                  futureCall,
+	gorums.E_Correctable:               correctableCall,
+	gorums.E_CorrectableStream:         correctableStreamCall,
+	gorums.E_Multicast:                 multicastCall,
+	strictordering.E_StrictOrderingQc:  strictOrderingQC,
+	strictordering.E_StrictOrderingRpc: strictOrderingRPC,
 }
 
 var gorumsCallTypeNames = map[*protoimpl.ExtensionInfo]string{
-	gorums.E_Qc:                "quorum",
-	gorums.E_QcFuture:          "asynchronous quorum",
-	gorums.E_Correctable:       "correctable quorum",
-	gorums.E_CorrectableStream: "correctable stream quorum",
-	gorums.E_Multicast:         "multicast",
+	gorums.E_Qc:                        "quorum",
+	gorums.E_QcFuture:                  "asynchronous quorum",
+	gorums.E_Correctable:               "correctable quorum",
+	gorums.E_CorrectableStream:         "correctable stream quorum",
+	gorums.E_Multicast:                 "multicast",
+	strictordering.E_StrictOrderingQc:  "strict ordering quorum",
+	strictordering.E_StrictOrderingRpc: "strict ordering",
+}
+
+// mapping from strict ordering type to a checker that will check if a method has that type
+var strictOrderingTypeCheckers = map[*protoimpl.ExtensionInfo]func(*protogen.Method) bool{
+	strictordering.E_StrictOrderingQc: func(m *protogen.Method) bool {
+		return hasAllMethodOption(m, gorums.E_StrictOrdering, gorums.E_Qc)
+	},
+	strictordering.E_StrictOrderingRpc: func(m *protogen.Method) bool {
+		return hasMethodOption(m, gorums.E_StrictOrdering) && !hasMethodOption(m, gorumsCallTypes...)
+	},
```

Comment:
This is a neat way to organize the various checkers that we need; to that end, I'm wondering if we should add more checkers here or in a separate map, e.g.:
```go
	gorums.E_Multicast: func(m *protogen.Method) bool {
		return hasMethodOption(m, gorums.E_Multicast) && m.Desc.IsStreamingClient()
	},
... and so on ...
```
If we do something like this, maybe we can probably avoid some of the checks in the switch cases in `validateMethodExtensions`... 

Any thoughts?